AppConnectSwift/CaptureImageViewController.swift

```diff
@@ -56,55 +54,29 @@ class CaptureImageViewController: UIViewController, UIImagePickerControllerDeleg
-                let clientFactory = MDClientFactory.sharedInstance()
-                let client = clientFactory.clientOfType(MDClientType.Network);
-                var datastore = MDDatastoreFactory.create()
-                var subject = datastore.subjectWithID(self.subjectID)
-                
-                subject.collectData(self.data, withMetadata: "Random String", contentType: "image/jpeg", completion: { (dataEnvelope:  MDSubjectDataEnvelope!, err: NSError!) -> Void in
-                    if err == nil {
-                        client.sendEnvelope(dataEnvelope, completion: { (err) in
-                            if err == nil {
-                                NSOperationQueue.mainQueue().addOperationWithBlock {
-                                    self.showAlert("Data saved successfully", message: "")
-                                    self.imageView.image = nil
-                                    subject = nil
-                                    bgQueue = nil
-                                    datastore = nil
-                                }
-                            }
-                            else if err.description.containsString("The Internet connection appears to be offline"){
-                                NSOperationQueue.mainQueue().addOperationWithBlock {
-                                    self.showAlert("Not connected to the Internet", message: "Please restore Internet connection and try again")
-                                    bgQueue = nil
-                                    datastore = nil
-                                }
-                            }
-                            else {
-                                NSOperationQueue.mainQueue().addOperationWithBlock {
-                                    self.showAlert("Unable to save data", message: err.description)
-                                    bgQueue = nil
-                                    datastore = nil
-                                }
-                            }
-                        })
-                    }
-                    else{
-                        self.showAlert("", message: err.description)
-                        bgQueue = nil
-                        datastore = nil
-                        subject = nil
-                    }
-                })
-            }
-        }
-        else {
+        
+        if self.data == nil {
             showAlert("No image selected", message: "")
+            return
         }
+        
+        let datastore = (UIApplication.sharedApplication().delegate as! AppDelegate).UIDatastore!
+        let subject = datastore.subjectWithID(self.subjectID)
+        
+        subject.collectData(self.data, withMetadata: "Random String", contentType: "image/jpeg", completion: { (err: NSError!) -> Void in
+            if (err == nil) {
+                
+                // update the UI.
+                dispatch_async(dispatch_get_main_queue()) {
```

Comment:
I don't have a preference there. I remember thinking dispatch was cleaner at the time... but check out this swift operator :

https://thatthinginswift.com/background-threads/